src/WPF/ArcGISRuntime.WPF.Samples/Samples/Hydrography/EncDisplaySettings/EncDisplaySettings.xaml.cs

```diff
@@ -0,0 +1,119 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using ArcGISRuntime.Samples.Managers;
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Hydrography;
+using Esri.ArcGISRuntime.Mapping;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace ArcGISRuntime.WPF.Samples.EncDisplaySettings
+{
+    public partial class EncDisplaySettings
+    {
+        public EncDisplaySettings()
+        {
+            InitializeComponent();
+
+            // Create the UI, setup the control references and execute initialization
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Apply initial display settings
+            UpdateDisplaySettings();
+
+            // Initialize the map with an oceans basemap
+            MyMapView.Map = new Map(Basemap.CreateOceans());
+
+            // Get the path to the ENC Exchange Set
+            string encPath = await GetEncPath();
+
+            // Create the Exchange Set
+            // Note: this constructor takes an array of paths because so that update sets can be loaded alongside base data
+            EncExchangeSet _encExchangeSet = new EncExchangeSet(new string[] { encPath });
+
+            // Wait for the exchange set to load
+            await _encExchangeSet.LoadAsync();
+
+            // Store a list of data set extent's - will be used to zoom the mapview to the full extent of the Exchange Set
+            List<Envelope> dataSetExtents = new List<Envelope>();
+
+            // Add each data set as a layer
+            foreach (EncDataSet _encDataSet in _encExchangeSet.DataSets)
+            {
+                // Create the cell and layer
+                EncLayer _encLayer = new EncLayer(new EncCell(_encDataSet));
+
+                // Add the layer to the map
+                MyMapView.Map.OperationalLayers.Add(_encLayer);
+
+                // Wait for the layer to load
+                await _encLayer.LoadAsync();
+
+                // Add the extent to the list of extents
+                dataSetExtents.Add(_encLayer.FullExtent);
+            }
+
+            // Use the geometry engine to compute the full extent of the ENC Exchange Set
+            Envelope fullExtent = GeometryEngine.CombineExtents(dataSetExtents);
+
+            // Set the viewpoint
+            MyMapView.SetViewpoint(new Viewpoint(fullExtent));
+        }
+
+        private async Task<String> GetEncPath()
+        {
+            #region offlinedata
+
+            // The data manager provides a method to get the folder
+            string folder = DataManager.GetDataFolder();
+
+            // Get the full path
+            string filepath = Path.Combine(folder, "SampleData", "AddEncExchangeSet", "ExchangeSet", "ENC_ROOT", "CATALOG.031");
```

Comment:
Can we add some comments on what the parameters of the Path.Combine mean? Perhaps something like:
// The resulting path will look something like:
// "\SampleData\AddEncExchangceSet\ExchangeSet\ENC_ROOT\Catalog.031"